_source/_posts/2022-02-07-cookies-vs-tokens.adoc

```diff
@@ -0,0 +1,135 @@
+
+- **Time**: The authentication token assigned to the user is only valid for a specific period of time. If the token becomes invalid, the user needs to be re-authenticated before access can be granted.
+
+NOTE: The term "token/authentication token" is used here to describe any form of authentication credential that might be implemented by the application, not explicitly an access token.
+
+The above instances aren't environment-specific and could be demonstrated visually on the frontend or the backend during API testing.
+
+== Why use authentication
+
+There are two reasons why authentication is necessary to allow full access to an application:
+
+- Authentication establishes the identity of the user and verifies that the user is who (or what) it says it is. Authentication protects your resources by denying access to unauthenticated users.
+
+- Authentication gives each user a distinct identity, protecting your data and theirs. Applications that require users to create an account give each user a unique profile, which is what determines the data shown to the user. For example, PayPal requires users to sign in before displaying their account balance and transactions.
+
+Generally, the process works like this:
+
+- You sign in.
+- The server verifies your sign-in details and assigns you an authentication token.
+- The authentication token is used to make a request to your homepage that displays your unique dashboard.
+
+Both session cookies and access tokens allow users to make requests to the server without needing to re-authenticate at each request. Following is a comparison of the two.
+
+== What you should know about cookies
+
+Session cookies are stateful elements. They contain data that the server sends to the browser for temporary use. The authentication data inside a cookie is stored on both the client and server. The server keeps track of active sessions in a database, while the browser holds the identifier to the active session. When a request is made to the server, the session ID is used to look up user information such as their roles or privileges for authentication, in order to check if the session is still valid.
+
+=== Advantages of cookies
+
+**Cookies provide the following benefits. They:**
+
+* **Use the same session across subdomains**: Cookies take a https://en.wikipedia.org/wiki/HTTP_cookie#Domain_and_Path[`Domain`] argument: You specify the domain name for which the cookie is valid. Setting the domain name to `yoursite.com` allows the same session for the domain and subdomains.
+
+* **Reduce manipulation by client-side JavaScript**: Cookies allow you to restrict client-side access by setting the https://owasp.org/www-community/HttpOnly[`HttpOnly`] flag. This reduces the likelihood of cross-site scripting (XSS) attacks on your application, since most XSS attacks involve the use of malicious JS code.
```

Comment:
```suggestion
* **Reduce manipulation by client-side JavaScript**: They allow you to restrict client-side access by setting the https://owasp.org/www-community/HttpOnly[`HttpOnly`] flag. This reduces the likelihood of cross-site scripting (XSS) attacks on your application, since most XSS attacks involve the use of malicious JS code.
```
replace the repetition of "Cookies this, cookies that..."